### DISCUSSION TITLE
High-impact destructive delete pattern

### DISCUSSION THREAD

**Discussion by vesnadean** [grafana/grafana#56199] [Saga Design System]
When users need to perform a delete action that is highly destructive, we want them to stop and think twice about whether they want to proceed, instead of having them mindlessly clicking okay. 

We currently have this pattern in Storybook, which achieves the above goals, but is case sensitive and most likely not i18n compliant.  
<img width="505" alt="Screenshot 2022-10-03 at 17 23 14" src="https://user-images.githubusercontent.com/74137331/193615258-bd3a7de6-c285-45a3-8154-f48455af8ebe.png">

The Cloud auth team came up with this pattern that uses a toggle component. The default state is "No" and users need to toggle it to "Yes". 
<img width="666" alt="Screenshot 2022-10-03 at 18 27 35" src="https://user-images.githubusercontent.com/74137331/193629827-4d2bef9d-00d7-40e0-9785-76057d92af75.png">

The question is:
1. Is the typing pattern annoying and does it fail internationalisation and localisation standards?
2. Is the toggle action enough to make users stop and think before performing a destructive action?
3. Should we tweak the existing pattern / come up with a new one?

**Comment by staton-hyse11** [comment]
@joshhunt do you have any insights on how i18n might come into play here? 

@VesnaDean If we remove the capital "D" from "Delete", do you think this would solve some of the concerns with this pattern? That would be a simple adjustment with little dev/design cost.

**Reply by vesnadean** [reply]
@staton-hyse11  It would definitely solve a part of the problem.

**Reply by joshhunt** [reply]
The typing pattern is kind of intentionally supposed to be annoying. Purposefully introducing friction so the user slows down and hopefully acknowledges the severity of their action. We can localise the word, but I think I would recommend not to - it just simplifies things and makes it more resistant to bugs. It doesn't matter exactly what they type, as long as it's something (and we tell them what to type)

Public Dashboards has a similar "speed hump" pattern:

![image](https://user-images.githubusercontent.com/46142/195627047-e3332641-abb5-436b-b103-ea3c0c564a3e.png)

**Reply by joshhunt** [reply]
I think the Toggle component is wrong to use here - for boolean yes/no options we should use checkbox instead.

**Reply by staton-hyse11** [reply]
I agree with the checkbox suggestion and it would align with the shared dashboard implementation. To me, the question that needs to be answered is: How much friction is appropriate for various destructive patterns? This needs alignment across the product and defined to ensure consistency.

I am leaning towards maintaining this pattern as is today and we can work to define these tiers of friction to better suit needs across the product in the meantime.

**Reply by staton-hyse11** [reply]
I've begun to capture some of the existing patterns in a delete pattern doc

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
